cpp/src/join/conditional_join.cu

```diff
@@ -118,8 +132,14 @@ conditional_join(table_view const& left,
 
   // If the output size will be zero, we can return immediately.
   if (join_size == 0) {
-    return std::make_pair(std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr),
-                          std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr));
+    auto join_indices{
+      std::make_pair(std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr),
+                     std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr))};
+    if (join_type == join_kind::FULL_JOIN) {
+      auto complement_indices = detail::get_left_join_indices_complement(
+        join_indices.second, left.num_rows(), right.num_rows(), stream, mr);
+      join_indices = detail::concatenate_vector_pairs(join_indices, complement_indices, stream);
```

Comment:
Wait, I see that `join_indices` is a local variable and it is not used anywhere?

Reply:
Previously there is a `return` statement here but it has been deleted. Is this intentional?

Reply:
Good catch, this is a mistake on my part. Fortunately (or maybe unfortunately?) the code still generates the correct result without it, it just does more work first. Fixed.

Reply:
I realized that `nvcc` does not catch all warnings as g++, which eventually be treated as errors so we can detect the  issue like this immediately. A lot of other ignored warnings were just recently fixed.